### PR TITLE
[Ethan] Rails 3.2.3 support

### PR DESCRIPTION
**Pull request by treybean**
https://www.pivotaltracker.com/story/show/28035853
https://www.pivotaltracker.com/story/show/28036207
https://www.pivotaltracker.com/story/show/28160117

### CODE REVIEW

File: app/models/tandem/content.rb [hunk 1]
```diff
@@ -3,9 +3,12 @@ class Content < ActiveRecord::Base
     validates :tag, presence: true
     validates :request_key, presence: true, uniqueness: {:scope => [:tag, :type]}
 
+    
     #enforce abstract class architecture
     validates :type, presence: true, exclusion: ['Tandem::Content']
 
+    attr_accessible :tag, :content, :details, :link_url, :link_target, :attachment_id, :attachment_type, :image_id, :request_key
```
Comment (evizitei) [inline app/models/tandem/content.rb:10]: hardcore security!!

File: app/views/tandem/images/_form.html.slim [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #tandem-upload-image
-  = form_for (@image || ::Tandem::Image.new), :html => { :multipart => true, :id => 'new_tandem_image' } do |f|
+  = form_for (@image ||= ::Tandem::Image.new), :url => (@image.new_record? ? tandem.images_path : tandem.image_path(@image)), :html => { :multipart => true, :id => 'new_tandem_image' } do |f|
```
Comment (evizitei) [inline app/views/tandem/images/_form.html.slim:2]: This works just fine, but it's kind of a long source line...maybe calculating the url a line above or something...or mabybe it's fine as is...I'm half and half

File: app/views/tandem/pages/_form.html.slim [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // This form controls New Page, Edit Page
 #tandem-editor
-  = form_for @page, remote: true, html: { class: 'tandem-page-form' } do |f|
+  = form_for @page, url: (@page.new_record? ? tandem.pages_path : tandem.page_path(@page)), remote: true, html: { class: 'tandem-page-form' } do |f|
```
Comment (evizitei) [inline app/views/tandem/pages/_form.html.slim:3]: If this happens a third time somewhere else, I will have to insist that we give tandem some kind of meta-programmed paths helper that does this check automatically :)

File: spec/dummy/config/application.rb [hunk 4]
```diff
@@ -48,6 +48,12 @@ class Application < Rails::Application
     config.assets.version = '1.0'
 
     config.action_mailer.default_url_options = { :host => 'localhost:3000' }
+
+    # Enforce whitelist mode for mass assignment.
+    # This will create an empty whitelist of attributes available for mass-assignment for all models
+    # in your app. As such, your models will need to explicitly whitelist or blacklist accessible
+    # parameters by using an attr_accessible or attr_protected declaration.
+    config.active_record.whitelist_attributes = true
```
Comment (evizitei) [inline spec/dummy/config/application.rb:56]: nice, no half measures for tandem

**Comment by evizitei** [comment]
No show stoppers, I'm merging

File: app/views/tandem/images/_form.html.slim (hunk 2 above)
Comment (treybean) [inline app/views/tandem/images/_form.html.slim:2]: Agreed. Chored.

File: app/views/tandem/pages/_form.html.slim (hunk 3 above)
Comment (treybean) [inline app/views/tandem/pages/_form.html.slim:3]: Or rather, this is chored.

**Pull request merged**